bundles/org.openhab.core.addon/src/main/java/org/openhab/core/addon/internal/xml/AddonInfoAddonsXmlProvider.java

```diff
@@ -79,37 +80,34 @@ public Set<AddonInfo> getAddonInfos(@Nullable Locale locale) {
     }
 
     private void initialize() {
-        File fileFolder = new File(folder);
+        File file = new File(fileName);
         try {
-            if (!fileFolder.isDirectory()) {
-                logger.debug("Folder '{}' does not exist", folder);
+            if (!file.isFile()) {
+                logger.warn("File '{}' does not exist.", fileName);
```

Comment:
IMO a missing file should not result in a warning. Distributions could decide to omit this file. A warning should be logged if it is a directory or the file cannot be read.